profiler/src/test/java/com/splunk/opentelemetry/profiler/JfrContextStorageTest.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright Splunk Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.splunk.opentelemetry.profiler;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.fail;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.ArgumentMatchers.isA;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.splunk.opentelemetry.profiler.events.ContextAttached;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.SpanContext;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.ContextStorage;
+import io.opentelemetry.context.Scope;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+class JfrContextStorageTest {
+
+  @Mock Span span;
+  @Mock ContextStorage delegate;
+  @Mock Context newContext;
+  @Mock Scope delegatedScope;
+  @Mock Function<Context, Span> spanFromContext;
+  @Mock SpanContext spanContext;
```

Comment:
Do you actually need all these mocks? Context API is totally usable by itself, no?

Reply:
* span - mocked in order to control whether or not the span context is in/valid
* delegate - mocked in order to return the delegated scope (below) so that we can assert that close is/isn't called
* newContext - the implementation of the Context interface is the ArrayBasedContext which has package access and cannot be instantiated, hence the mock
* delegatedScope - mocked so that we can verify that close() is/isn't called
* spanFromContext - this could actually be a real implementation but would probably add some code.  is there a specific problem with this being a mock?
* spanContext - mocked so that we can control whether or not the context is/isn't valid.

I don't know how to use the existing Context API by itself to drive this behavior without mocks.

Reply:
As long as you're not doing `context.makeCurrent()` or anything to do with the scope/storage, you should just be able to work with the Context objects as POJOs. So, you can just do `Context.root().with(span)` etc.

Reply:
And `SpanContext` has a `getInvalid()` method on it to get an invalid one without a mock, as well.

Reply:
Ok, I was able to simplify things a little. Thanks.